controllers/src/main/scala/cromwell/pipeline/controller/RunController.scala

```diff
@@ -0,0 +1,57 @@
+package cromwell.pipeline.controller
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Directives.{ entity, _ }
+import akka.http.scaladsl.server.Route
+import cromwell.pipeline.datastorage.dto.auth.AccessTokenContent
+import cromwell.pipeline.datastorage.dto.{ RunCreateRequest, RunDeleteRequest, RunUpdateRequest }
+import cromwell.pipeline.model.wrapper.RunId
+import cromwell.pipeline.service.RunService
+import de.heikoseeberger.akkahttpplayjson.PlayJsonSupport._
+
+import scala.concurrent.ExecutionContext
+import scala.util.{ Failure, Success }
+
+class RunController(runService: RunService)(implicit executionContext: ExecutionContext) {
+
+  val route: AccessTokenContent => Route = accessToken =>
+    path("runs") {
+      concat(
+        get {
+          parameter('run_id.as[String]) { runId =>
+            {
+              onComplete(runService.getRunByIdAndUser(RunId.from(runId), accessToken.userId)) {
```

Comment:
`runService.getRunByIdAndUser` must take `RunId` (as the name implies) as a parameter, not `ValidationResult[Wrapper]`.
You can leverage deserialization to akka by changing
`parameter('run_id.as[String]) { runId =>`
to
`parameter('run_id.as[RunId]) { runId =>`
In that case, you will have to provide an implicit Unmarshaller.
Alternatively, you can pattern match on the result of the `RunId.from` right here, or you must make a method in a `runService` with a more appropriate name.

Reply:
I created a Unmarshaller function. I am not sure if this needs to be placed there

Reply:
It should be placed somewhere nearby, for example `cromwell.pipeline.controller.utils.FromStringUnmarshallers`. Also, if you make it implicit (e.g. `implicit val stringToRunId = ???`) then `'run_id.as[RunId]` will work as expected.

Reply:
I placed the function in the path that you suggested. Also, the implicit identifier. I am really amazed of how scala can treat this, happy to learn things about this.Thanks!